### PR TITLE
Store vespa version in ZooKeeper as well

### PR DESCRIPTION
**Pull request by hmusum**
Use version in ZooKeeper if configserver-distribute-application-package is true,
fallback to version in file. Will always write to ZooKeeper so this can be
the only source in the future.

This means that if configserver-distribute-application-package is true
and version data is found in ZooKeeper, redeployment of appplication packages
will only be done on one config server.

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java [hunk 1]
```diff
@@ -5,45 +5,78 @@
...
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
     public Version storedVersion() {
+        if (distributeApplicationPackage.value()) {
+            Optional<byte[]> version = curator.getData(versionPath);
+            if(version.isPresent()) {
+                System.out.println("Found version in zk ");
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java:70]: Remove or log

File: configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java [hunk 2]
```diff
@@ -5,45 +5,78 @@
...
+                try {
+                    return Version.fromString(Utf8.toString(version.get()));
+                } catch (Exception e) {
+                    return new Version(0, 0, 0); // Use an old value to signal we don't know
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java:74]: Consider falling back to the file.

File: configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java [hunk 3]
```diff
@@ -5,45 +5,78 @@
...
+                }
+            }
+        }
         try (FileReader reader = new FileReader(versionFile)) {
+            System.out.println("Found version in file ");
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java:79]: Remove or log

File: configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java (hunk 2 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java:74]: Good point, fixed

File: configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java:70]: Fixed

File: configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java (hunk 3 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/version/VersionState.java]: Fixed

**Review by hakonhall — APPROVED** [review]

**Pull request merged**